File: 0021-MergeTwoSortedLists.mjs
import LinkedList from "./LinkedListUtils.mjs";
import { ListNode } from "./LinkedListUtils.mjs";

var mergeTwoLists = function (list1, list2) {
  const head = new ListNode('head');
  let next_head = head;
  while (list1 && list2) {

    if (list1.val <= list2.val) {
      next_head.next = list1;
      list1 = list1.next;
    } else {
      next_head.next = list2;
      list2 = list2.next;
    }
    next_head = next_head.next;
  }
  if (list1) {
    next_head.next = list1;
  }
  if (list2) {
    next_head.next = list2;
  }
  return head.next;
};

const LL = new LinkedList();
const list1 = LL.createLinkedList([]);
const list2 = LL.createLinkedList([1, 3, 4]);
const rtn = mergeTwoLists(list1, list2);
LL.printLinkedList(rtn);

File: 0021-MergeTwoSortedLists.py
from typing import Optional
from LinkedListUtil import ListNode


class Solution:
    """ solution class """

    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        """Method to combine two, ordered linked lists into one"""
        head = ListNode()
        next_head = head
        while list1 and list2:
            if list1.val <= list2.val:
                next_head.next = list1
                list1 = list1.next
            else:
                next_head.next = list2
                list2 = list2.next
            next_head = next_head.next
        
        if (list1):
            next_head.next = list1

        if (list2):
            next_head.next = list2

        return head.next
